variant-remapping-automation/remapping_process.nf

```diff
@@ -0,0 +1,137 @@
+
+
+process retrieve_source_genome {
+
+    output:
+    path "${params.source_assembly_accession}.fa" into source_fasta
+    path "${params.source_assembly_accession}_assembly_report.txt" into source_report
+
+    """
+    $params.executable.genome_downloader --assembly-accession ${params.source_assembly_accession} --species ${species_name} --output-directory ${params.genome_assembly_dir}
+    ln -s ${params.genome_assembly_dir}/${species_name}/${params.source_assembly_accession}/${params.source_assembly_accession}.fa
+    ln -s ${params.genome_assembly_dir}/${species_name}/${params.source_assembly_accession}/${params.source_assembly_accession}_assembly_report.txt
+    """
+}
+
+
+process retrieve_target_genome {
+
+    output:
+    path "${params.target_assembly_accession}.fa" into target_fasta
+    path "${params.target_assembly_accession}_assembly_report.txt" into target_report
+
+    """
+    $params.executable.genome_downloader --assembly-accession ${params.target_assembly_accession} --species ${species_name} --output-directory ${params.genome_assembly_dir}
+    ln -s ${params.genome_assembly_dir}/${species_name}/${params.target_assembly_accession}/${params.target_assembly_accession}.fa
+    ln -s ${params.genome_assembly_dir}/${species_name}/${params.target_assembly_accession}/${params.target_assembly_accession}_assembly_report.txt
+    """
+}
+
+
+/*
+ * Extract the submitted variants to remap from the accesioning warehouse and store them in a VCF file.
+ */
+process extract_vcf_from_mongo {
+    input:
+    path source_fasta from source_fasta
+    path source_report from source_report
+
+    output:
+    path "${params.source_assembly_accession}_extraction.properties" into extraction_props
+    // Store both vcfs (eva and dbsnp) into one channel
+    path "*.vcf" into source_vcfs
+    val "${params.source_assembly_accession}_vcf_extractor.log" into log_filename
+
+    """
+    cp ${params.template_properties} ${params.source_assembly_accession}_extraction.properties
+    echo "parameters.fasta=${source_fasta}" >> ${params.source_assembly_accession}_extraction.properties
+    echo "parameters.assemblyReportUrl=file://${source_report}"" >> ${params.source_assembly_accession}_extraction.properties
+    echo "parameters.assemblyAccession=${params.source_assembly_accession}"" >> ${params.source_assembly_accession}_extraction.properties
+    echo "parameters.outputFolder=." >> ${params.source_assembly_accession}_extraction.properties
+
+    java -jar $params.jar.vcf_extractor --spring.config.name=${params.source_assembly_accession}_extraction.properties > ${params.source_assembly_accession}_vcf_extractor.log
+    """
+}
+
+
+/*
+ * variant remmapping pipeline
+ */
+process remap_variants {
+
+    input:
+    path source_fasta from source_fasta
+    path target_fasta from target_fasta
+    path source_vcf from source_vcfs
+
+
+    output:
+    path "remapped_vcf.vcf" into remapped_vcfs
+    path "remapped_vcf.vcf.yml" into remapped_ymls
```

Comment:
is this yaml file being used?

Reply:
It now is being output to the publishDir